### DISCUSSION TITLE
How to export and import a virtual machine

### DISCUSSION THREAD

**Discussion by tocha688** [quickemu-project/quickemu#1400] [Quickemu] [answered]
**I confirm this feature has not been previously requested**
- [ ] I have searched the issues and this feature has not previously been requested

**Is your feature request related to a problem? Please describe.**
I am a novice and I hope to be able to export and import the virtual machines I have created so that I can share them with my friends.
In addition, I don't know how to shut down and restart the virtual machine. Can you tell me? Thank you

**Comment by lj3954** [comment, marked as answer]
The VM is comprised of your configuration file and the VM directory, likely under the same name. You can move those directly or add them to a tarball.
Shutting down the VM directly through the guest is preferred, but if that's impossible for whatever reason, you can use the QEMU monitor through `quickemu --vm ${CONFIG} --monitor-cmd ${MONITOR_CMD}`. Some commands that may be useful for this case are `system_powerdown`, `system_reset`, and `quit`. The full documentation can be found at https://www.qemu.org/docs/master/system/monitor.html.

**Reply by tocha688** [reply]
OK, thank you for answering me.
